### PR TITLE
Adds SSL support

### PR DESCRIPTION
**Pull request by keaner**
Adds serveSecure method which passes through to bindSecure

### CODE REVIEW

**Comment by kevmoo** [comment]
I like it – although I wonder if we should just add `securityContext` as an optional argument to serve.

@nex3 ?

File: lib/shelf_io.dart [hunk 1]
```diff
@@ -57,6 +57,23 @@ void serveRequests(Stream<HttpRequest> requests, Handler handler) {
...
 
+/// Starts an SSL [HttpServer] that listens on the specified [address] and
+/// [port] and sends requests to [handler].
+///
+/// See the documentation for [HttpServer.bindSecure] for more details on
+/// [address], [port], and [backlog].
+Future<HttpServer> serveSecure(
+    Handler handler, address, int port, SecurityContext serverContext,
+    {int backlog}) {
+  if (backlog == null) backlog = 0;
```
Comment (jakemac53) [inline lib/shelf_io.dart]: nit: `backlog ??= 0`

**Comment by nex3** [comment]
> I like it – although I wonder if we should just add `securityContext` as an optional argument to serve.

I like this idea.

**Comment by keaner** [comment]
Cool, I'll make SecurityContext an optional arg and add tests

**Comment by kevmoo** [comment]
Awesome.

**Comment by keaner** [comment]
Updated based on recommendations and added two tests.

I originally tried to add the certs as separate files but I was running into this error:
"Unsupported operation: Cannot extract a file path from a data URI"

File: lib/shelf_io.dart [hunk 2]
```diff
@@ -30,11 +30,22 @@ export 'src/io_server.dart';
...
-  return HttpServer.bind(address, port, backlog: backlog).then((server) {
+/// If a [securityContext] is provided an HTTPS server will be started.
+////
+/// See the documentation for [HttpServer.bind] and [HttpServer.bindSecure]
+/// for more details on [address], [port], and [backlog].
+Future<HttpServer> serve(Handler handler, address, int port,
+    {SecurityContext securityContext, int backlog}) {
+  backlog ??= 0;
+  if (securityContext == null) {
+    return HttpServer.bind(address, port, backlog: backlog).then((server) {
```
Comment (nex3) [inline lib/shelf_io.dart]: Use async/await here rather than duplicating `then()` calls.

File: test/shelf_io_ssl_test.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        _serverPort = server.port;
+      }));
+}
+
+Future<HttpClientRequest> _scheduleGet() {
+  return schedule/*<Future<HttpClientRequest>>*/(() {
+    return client.getUrl(Uri.parse('https://localhost:$_serverPort/'));
+  });
+}
```
Comment (nex3) [inline test/shelf_io_ssl_test.dart]: Can you make this part of `shelf_io_test.dart`?

File: lib/shelf_io.dart (hunk 2 above)
Comment (keaner) [inline lib/shelf_io.dart]: Done!

File: test/shelf_io_ssl_test.dart (hunk 3 above)
Comment (keaner) [inline test/shelf_io_ssl_test.dart]: Done!

**Pull request merged**